src/main/java/com/zaxxer/nuprocess/internal/BaseEventProcessor.java

```diff
@@ -128,6 +131,7 @@ public void shutdown()
          process.onExit(Integer.MAX_VALUE - 1);
          LibC.waitpid(process.getPid(), exitCode, LibC.WNOHANG);
       }
+      ((Memory) exitCode.getPointer()).close();
```

Comment:
I don't really like this assumption about the internals of JNA. You're downcasting to a derived type (`Memory` extends `Pointer`), which, _per the API_, may fail, and you're doing it in an unchecked way (as in, if the cast fails, you're not handling it)--all to free 4 bytes of memory that will end up automatically freed at some point without your intervention. Especially for the set of `ProcessEpoll` threads that handle `start`ed processes, which only terminate when the JVM does, the benefit of this cross-domain trespass feels effectively nonexistent.

Given `Memory` implements `Closeable`, if we want to do this, I wonder if we should consider adding a simple helper method somehwere in this `com.zaxxer.nuprocess.internal` package along the lines of:
```java
public static void closeIfCloseable(Object object) {
    if (object instanceof Closeable) {
        ((Closeable) object).close();
    }
}
```

Then all the different places you do this cast (and there look to be several), you can use the helper instead.

Reply:
Spot on. I didn't like it either. I just added a generic sounding Util method with close methods.